### PR TITLE
Tests for run-remote-query

### PR DESCRIPTION
**Pull request by shati-patel**
First pass at adding tests:

- Tests for `getRepositories` ✔️ 
- Tests for ~`validateRepositories`~ `attemptRerun` ✔️ 
- Tests for `runRemoteQuery`: still TODO (I can follow-up in separate PR)

There's still more to test, but (to avoid this PR getting too big) it's probably helpful to get this reviewed and merged so far...

I've tried testing the main interesting cases for `getRepositories` and `attemptRerun`. Is there anything else that would be helpful to test? (Or not test? 🙃) 

## Checklist

N/A

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] `@github/docs-content-codeql` has been cc'd in all issues for UI or other user-facing changes made by this pull request.

### CODE REVIEW

File: extensions/ql-vscode/src/run-remote-query.ts [hunk 1]
```diff
@@ -130,33 +130,38 @@ async function runRemoteQueriesApiRequest(credentials: Credentials, ref: string,
...
+    }
 
-    } else {
-      void showAndLogErrorMessage(error);
+    const popupMessage = 'Unable to run query on some of the specified repositories. [See logs for more details](command:codeQL.showLogs).';
+    const rerunQuery = await showInformationMessageWithAction(popupMessage, 'Rerun on the valid repositories only');
+    if (rerunQuery) {
+      const validRepositories = repositories.filter(r => !invalidRepos.includes(r) && !reposWithoutDbUploads.includes(r));
+      void logger.log(`Rerunning query on set of valid repositories: ${JSON.stringify(validRepositories)}`);
+      await runRemoteQueriesApiRequest(credentials, ref, language, validRepositories, query);
```
Comment (aeisenberg) [inline extensions/ql-vscode/src/run-remote-query.ts:160]: The function is named `validateRepositories`, but here it is re-running the request. Either this function's name should be changed to reflect what it is doing, or this call should be removed from the function.

Probably simpler to just rename. How about something like `rerunFailedRepositories`? Or `attemptRerun`?

**Review by aeisenberg — COMMENTED** [review]
Great work! 

One small comment and it's good to go.

File: extensions/ql-vscode/src/run-remote-query.ts (hunk 1 above)
Comment (shati-patel) [reply extensions/ql-vscode/src/run-remote-query.ts:161]: Good point 😄 `attemptRerun` is more descriptive 👍🏽

**Review by aeisenberg — APPROVED** [review]
Nice job!

**Comment by shati-patel** [comment]
> Nice job!

Thanks for all the help 😊

**Pull request merged**